src/Config/ObjectModel/NestedInsertOptions.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Azure.DataApiBuilder.Config.ObjectModel;
+
+/// <summary>
+/// Options for nested create operations.
+/// </summary>
+/// <param name="Enabled"> Indicates whether nested create operation is enabled.</param>
+public class NestedCreateOptions
```

Comment:
Why not use a more generic name like `NestedMutationOptions`? Makes more sense when we start supporting nested updates as well?